netclient/wireguard/mac.go

```diff
@@ -0,0 +1,273 @@
+package wireguard
+
+import (
+	"bufio"
+	"errors"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gravitl/netmaker/models"
+	"github.com/gravitl/netmaker/netclient/ncutils"
+)
+
+// WgQuickDownMac - bring down mac interface, remove routes, and run post-down commands
+func WgQuickDownMac(node models.Node, iface string) error {
+	if err := RemoveConfMac(iface); err != nil {
+		return err
+	}
+	if node.PostDown != "" {
+		runcmds := strings.Split(node.PostDown, "; ")
+		ncutils.RunCmds(runcmds, true)
+	}
+	return nil
+}
+
+// RemoveConfMac - bring down mac interface and remove routes
+func RemoveConfMac(iface string) error {
+	var err error
+	realIface, err := getRealIface(iface)
+	if realIface != "" {
+		err = deleteInterface(iface, realIface)
+	}
+	return err
+}
+
+// WgQuickUpMac - bring up mac interface and set routes
+func WgQuickUpMac(node models.Node, iface string, confPath string) error {
+	var err error
+	var realIface string
+	realIface, err = getRealIface(iface)
+	if realIface != "" && err == nil {
+		deleteInterface(iface, realIface)
+		deleteRoutes(realIface)
+	}
+	realIface, err = addInterface(iface)
+	if err != nil {
+		ncutils.PrintLog("error creating wg interface", 1)
+		return err
+	}
+	time.Sleep(time.Second / 2)
+	err = setConfig(realIface, confPath)
+	if err != nil {
+		ncutils.PrintLog("error setting config for "+realIface, 1)
+		return err
+	}
+	var ips []string
+	ips = append(node.AllowedIPs, node.Address)
```

Comment:
could do:
var ips = append(node.AllowedIPs, []string{node.Address, node.Address6}...) if you wanna get to 1 line